service/sidecar/mapping.go

```diff
@@ -242,7 +242,7 @@ func verifyTxForm(tx *protoblocktx.Tx) protoblocktx.Status {
 	if len(tx.Namespaces) == 0 {
 		return protoblocktx.Status_MALFORMED_EMPTY_NAMESPACES
 	}
-	if len(tx.Namespaces) != len(tx.Signatures) {
+	if len(tx.Namespaces) != len(tx.Endorsements) {
```

Comment:
major: We should also check here that if we use "threshold" endorsement, it has exactly one endorsement.